src/plcc/load_spec/validate_spec/validate_lexical_spec_test.py

```diff
@@ -0,0 +1,151 @@
+    secondValidName = makeLexicalRule(makeLine("VALID_2"), False, "VALID_2", "+")
+    lexicalSpec = makeLexicalSpec([validName, secondValidName])
+    errors = validate_lexical_spec(lexicalSpec)
+    assert len(errors) == 0
+
+def test_line_invalid_rule_error():
+    line = makeLine("gibberish with no pattern or token")
+    lexicalSpec = makeLexicalSpec([line])
+    errors = validate_lexical_spec(lexicalSpec)
+    assert len(errors) == 1
+    assert errors[0] == makeInvalidRuleError(line)
+
+def test_rule_followed_by_line_invalid_rule_error():
+    rule = makeLexicalRule(makeLine("TEST \'\\w+\'"), False, "TEST", "\\w+")
+    line = makeLine("there is nothing useful here")
+    lexicalSpec = makeLexicalSpec([rule, line])
+    errors = validate_lexical_spec(lexicalSpec)
+    assert len(errors) == 1
+    assert errors[0] == makeInvalidRuleError(line)
+
+def test_closing_quotes_pattern_is_an_error():
+    closingQuotes = makeLexicalRule(makeLine("TEST \'\"\'"), False, "TEST", "\"")
+    lexicalSpec = makeLexicalSpec([closingQuotes])
+    errors = validate_lexical_spec(lexicalSpec)
+    assert len(errors) == 1
+    assert errors[0] == makeInvalidPatternError(closingQuotes)
+
+def test_closing_quotes_anywhere_in_pattern_is_an_error():
+    closingQuotes = makeLexicalRule(makeLine("TEST \'+\"\'"), False, "TEST", "+\"")
+    lexicalSpec = makeLexicalSpec([closingQuotes])
+    errors = validate_lexical_spec(lexicalSpec)
+    assert len(errors) == 1
+    assert errors[0] == makeInvalidPatternError(closingQuotes)
+
+def test_multiple_errors():
+    validName = makeLexicalRule(makeLine("NAME \'-\'"), False, "NAME", "-")
+    invalidName = makeLexicalRule(makeLine("name \'+\'"), False, "name", "+")
+    duplicateName = makeLexicalRule(makeLine("NAME \'\\s+\'"), False, "NAME", "\\s+")
+    line = makeLine("no rules here")
+    lexicalSpec = makeLexicalSpec([validName, invalidName, duplicateName, line])
+    errors = validate_lexical_spec(lexicalSpec)
+    assert len(errors) == 3
+    assert errors[0] == makeInvalidNameFormatError(invalidName)
+    assert errors[1] == makeDuplicateNameError(duplicateName)
+    assert errors[2] == makeInvalidRuleError(line)
+
+def makeLexicalSpec(ruleList=None):
+    return LexicalSpec(ruleList)
+
+def makeLexicalRule(line, isSkip, name, pattern):
+    return LexicalRule(line, isSkip, name, pattern)
+
+def makeLine(string, lineNumber=1, file=None):
+    return Line(string, lineNumber, file)
+
+def makeInvalidNameFormatError(rule):
+    message = f"Invalid name format for rule '{rule.name}' (Must be uppercase letters, numbers, and underscores, and cannot start with a number) on line: {rule.line.number}"
+    return makeValidationError(rule.line, message)
+
+def makeDuplicateNameError(rule):
+    message = f"Duplicate rule name found '{rule.name}' on line: {rule.line.number}"
+    return makeValidationError(rule.line, message)
+
+def makeDuplicatePatternError(rule):
```

Comment:
Coverage shows this function isn't used. If it was supposed to be used, use it; otherwise please delete it.